pdqsort.h

```diff
@@ -478,8 +508,19 @@ namespace pdqsort_detail {
             } else {
                 // If we were decently balanced and we tried to sort an already partitioned
                 // sequence try to use insertion sort.
-                if (already_partitioned && partial_insertion_sort(begin, pivot_pos, comp)
-                                        && partial_insertion_sort(pivot_pos + 1, end, comp)) return;
+                if (already_partitioned && partial_insertion_sort(begin, pivot_pos, comp)) {
+                    if(partial_insertion_sort(pivot_pos + 1, end, comp)) return;
+                    begin = pivot_pos + 1;
+                    leftmost = false;
+                    continue;
+                }
+
+            }
+
+            if((end - (pivot_pos + 1)) < insertion_sort_threshold) {
+                unguarded_insertion_sort(pivot_pos + 1, end, comp);
+                end = pivot_pos;
+                continue;
```

Comment:
I don't see a good reason to add another call to insertion sort here.

Reply:
The idea is to finish handling the small region while it's still hot (it also decreases recursive depth/stack usage)

Reply:
Well the call is definitely overhead, but I mean the data at [pivot+1, end). It's hot before the recursive call, after the recursion it may be out of L1.

Reply:
Conversely, adding the extra call can clobber the icache more.